webapp/src/ts/modules/messages/messages-content.component.ts

```diff
@@ -0,0 +1,322 @@
+  }
+
+  // See URL parameter "id" note at top of file
+  private getContactable(id, type) {
+    if (type === 'contact') {
+      return this.lineageModelGeneratorService
+        .contact(id)
+        .catch(err => {
+          if (err.code === 404) {
+            return;
+          }
+          throw err;
+        });
+    }
+
+    if (type === 'phone') {
+      return { name: id };
+    }
+
+    return {};
+  }
+
+  private markConversationReadIfNeeded() {
+    if (!this.selectedConversation) {
+      return;
+    }
+
+    const hasUnreadDoc = this.selectedConversation.messages.some(message => !message.read && message.doc);
+
+    if (hasUnreadDoc) {
+      const docs = this.selectedConversation.messages.map(message => message.doc);
+      this.markReadService
+        .markAsRead(docs)
+        .then(() => {
+          this.messagesActions.markSelectedConversationRead();
+        })
+        .catch(err => console.error('Error marking all as read', err));
+    }
+  }
+
+  selectContact(id, type) {
+    // See URL param "id" note at top of file.
+    if (!id) {
+      this.messagesActions.setMessagesError(false);
+      this.globalActions.setLoadingContent(false);
+      this.globalActions.unsetSelected();
+      return;
+    }
+
+    const refreshSelected = this.selectedConversation && this.selectedConversation.id === id;
+    this.messagesActions.setSelected({ id: id, messages: [] }, refreshSelected);
+    this.globalActions.setLoadingShowContent(id);
+
+    return Promise
+      .all([
+        this.getContactable(id, type),
+        this.messageContactService.getConversation(id)
+      ])
+      .then(([contactModel, conversation]) => {
+        if (!contactModel && conversation.length) {
+          const firstTaskWithContact = conversation[0].doc.tasks.find((task) => {
+            const message = task.messages && task.messages[0];
+            return message && message.contact && message.contact._id === id;
+          });
+
+          const firstMessageWithContact = firstTaskWithContact.messages.find(message => message.contact._id === id);
+          contactModel = {
+            doc: {
+              name: '',
+              phone: firstMessageWithContact.to
+            }
+          };
+        }
+
+        if (this.selectedConversation && this.selectedConversation.id !== id) {
+          // ignore response for previous request
+          return;
+        }
+
+        this.globalActions.setLoadingShowContent(false);
+        this.messagesActions.setMessagesError(false);
+        const unread = conversation.filter(message => !message.read);
+        this.firstUnread = _minBy(unread, message => message.doc.reported_date);
+        this.messagesActions.updateSelectedConversation({ contact: contactModel, messages: conversation });
+        this.globalActions.setTitle((contactModel && contactModel.doc && contactModel.doc.name) || id);
+        this.markConversationReadIfNeeded();
+        setTimeout(() => this.scrollToUnread());
```

Comment:
@dianabarsan this code is running in the last "init" hook, so the next hook is one of the "change" hook, which will execute scrollToUnread on every little change in the template. 
If we try to do this in the init hooks, there's a change that the scroll wont happen on time if the request to retrieve conversations is slow for some reason.
I tried to chain it with promises, for example: after retrieving conversation then scroll, but this isn't giving enough time for the elements to be in the template and the scroll to find it.
I tried async await the selectContact and then scroll, but same bad result.
So, maybe if we don't want timeout here, we can put the scroll function in the "change" hook to let it try and try until it finds the element and then make it stop executing with a flag or something.

Reply:
I think it's fine to execute in a change hook with a flag. 👍🏻 do you think it's worth the try?

Reply:
I tried to do that approach by using the 2 change hooks with a flag. It partially works, it's not scrolling all the way down, I think it's because it scrolls when it finds the wrapper container, but the content inside hasn't finished injecting all the items and grows a bit more.
I think we have 2 option: let it scroll by every template change (impacting performance), or use the timeout that I'm not too sure if it's going to be 100% reliable. I'd be nice to know when the content has finished rendering...

Reply:
I decided to leave the setTimeout to avoid adding on performance.

Reply:
Didn't the `setTimeout` trigger an application-wide change detection ? 
https://stackoverflow.com/a/42972499/1865409

Wouldn't that be the worst hit on performance?

Reply:
I'm still unsure if the native setTimeout really triggers an application-wide change detection, I need to read more on how they intercept the native setTimeout.
I have made few changes to avoid the setTimeout. I added a fixed min-height to the async pipes, used the afterViewChange and added a flag to avoid running more than necessary the scrolling. However, if something async is added and this modifies the container height, then the scrolling might be affected :(